[fix] Amend Input component, Login page, Signup page

CHAERYEON-LEE opened the pull request:
### 1️⃣ PR 타입(하나 이상의 PR 타입을 선택해주세요)
- [ ] 기능 추가
- [ ] 기능 삭제
- [X] 버그 수정
- [ ] 의존성, 환경 변수, 빌드 관련 코드 업데이트
<br>

### 2️⃣ 이슈 번호
ex) close #48  close #45 
close #47 
<br>
<br>

### 3️⃣ 변경 사항
로그인 페이지 수정
- state 추가
- 인풋 컴포넌트 수정에 다른 로그인 페이지의 인풋 컴포넌트 수정

인풋 컴포넌트 수정 및 회원가입 페이지 구현

배경색과 일치하는 버튼컴포넌트의 색상 변경
- background 색상을 color-main-orange-700에서 600으로 변경

인풋 컴포넌트 스타일 수정
- md, sm 사이즈일때 label 숨기기
- md 사이즈에서 border-radius 값 8px 로 수정, 테마에 특정 값이 없어서 px로 수정하였음

인풋 컴포넌트의 name props 타입 변경
- name props의 type을 '' 문자열로 변경하여 여러종류를 받을 수 있게 하였습니다.
- 스토리북의 설명을 참고하여 사용하고 싶은 인풋의 종류를 선택해서 사용할 수 있습니다.
- 만약 사용하고 싶은 종류가 없다면 원하는 문자열을 name 에 적으면 자동으로 type이 text로 적용됩니다.
<br>
<br>

CHAERYEON-LEE committed:
refactor: Amend signup page, input component

인풋 컴포넌트 수정 및 회원가입 페이지 구현

CHAERYEON-LEE committed:
fix: Amend Button component background color

배경색과 일치하는 버튼컴포넌트의 색상 변경
- background 색상을 color-main-orange-700에서 600으로 변경

CHAERYEON-LEE committed:
fix: Amend login page

로그인 페이지 수정
- state 추가
- 인풋 컴포넌트 수정에 다른 로그인 페이지의 인풋 컴포넌트 수정

CHAERYEON-LEE committed:
fix: Amend Input component style

인풋 컴포넌트 스타일 수정
- md, sm 사이즈일때 label 숨기기
- md 사이즈에서 border-radius 값 8px 로 수정, 테마에 특정 값이 없어서 px로 수정하였음

pereng11 commented on src/components/common/Input/Input.tsx:
```diff
@@ -24,15 +24,16 @@ export type InputProps = {
    * 인풋 밸류 스테이트 핸들러
    */
-  handler: (value: string) => void;
+  onChange: (value: any) => void;
```
```javascript
onChange: (e:React.ChangeEvent<HTMLInputElement>) => void
```
타입 정확하면 더 좋을 것 같습니다!

CHAERYEON-LEE replied:
넵!!

CHAERYEON-LEE committed:
fix: Amend Input component name props

인풋 컴포넌트의 name props 타입 변경
- name props의 type을 '' 문자열로 변경하여 여러종류를 받을 수 있게 하였습니다.
- 스토리북의 설명을 참고하여 사용하고 싶은 인풋의 종류를 선택해서 사용할 수 있습니다.
- 만약 사용하고 싶은 종류가 없다면 원하는 문자열을 name 에 적으면 자동으로 type이 text로 적용됩니다.

pereng11 reviewed (commented):
좋아요좋아요~
1. 버튼도 className 받을 수 있도록 부탁드립니다.
2. form에서 form.[inputName]으로 input요소에 접근하는 방법을 사용한다면 input의 value에 관한 상태를 Input 컴포넌트 내부에서 관리하는 것도 가능할 것 같아요. 
3. validation이 필요한 Input과 필요하지 않은 Input을 아예 다른 컴포넌트로 분리하는 것도 괜찮아 보입니다.

저는 MissionForm 만들면서 2번처럼 Input을 사용했는데 그것도 나쁘지 않아 보이더라구요. 노션에 관련 내용 정리해놨으니 시간되면 한 번 봐주시면 좋을 것 같습니다!

pereng11 commented on src/components/common/Input/Input.tsx:
```diff
@@ -3,10 +3,12 @@ import { useState, useId } from 'react';
+   * 자유롭게 입력한 문자열은 자동으로 인풋의 타입이 'text'로 지정됩니다.
    */
-  name: 'id' | 'password' | 'mission' | 'herocode' | 'confirm' | 'register';
+  name: '';
```
default props랑 헷갈리신건가요..? 이러면 name에 빈 문자열만 들어올 수 있습니다!

CHAERYEON-LEE replied:
string 으로 수정하겠습니다!!!

CHAERYEON-LEE commented:
> 좋아요좋아요~
> 
> 1. 버튼도 className 받을 수 있도록 부탁드립니다.
> 2. form에서 form.[inputName]으로 input요소에 접근하는 방법을 사용한다면 input의 value에 관한 상태를 Input 컴포넌트 내부에서 관리하는 것도 가능할 것 같아요.
> 3. validation이 필요한 Input과 필요하지 않은 Input을 아예 다른 컴포넌트로 분리하는 것도 괜찮아 보입니다.
> 
> 저는 MissionForm 만들면서 2번처럼 Input을 사용했는데 그것도 나쁘지 않아 보이더라구요. 노션에 관련 내용 정리해놨으니 시간되면 한 번 봐주시면 좋을 것 같습니다!
